PayPalDataCollector/src/main/java/com/braintreepayments/api/PayPalDataCollectorInternalRequest.java

```diff
@@ -7,17 +7,25 @@
 /**
  * Used to configuration the PayPalDataCollector request
  */
-class PayPalDataCollectorRequest {
+class PayPalDataCollectorInternalRequest {
```

Comment:
Instead of renaming this and making a new `PayPalDataCollectorRequest` class. Could we instead just only publicly expose the `hasUserLocationConsent` bool on here?

Reply:
I kinda like the clearer separation between internal and merchant-facing classes. Once we expose one field, we'll have a hard time ever refactoring this class if we need. I'm also not sure how the constructor would look once we convert the request class to Kotlin 🤔 

But definitely open to this if folks feel like it's the better route.

Reply:
I'm +1 for the sake of simplicity. Internal properties on `PayPalDataCollectorRequest` will result in less code since it's a Kotlin data class, which is a nice win for maintainability.

Reply:
I'm going to merge the PR in its current state but will open a separate PR into the feature branch if we find a solution that works 👍

Reply:
Ok cool. Yeah basically the top-level constructor is made internal, and the convenience constructor is made public. We do this a lot [on PPCP](https://github.com/paypal/paypal-android/blob/cf95ac7982b529f3caf25afd901e5687dfd8e810/PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt#L14).

Reply:
Thanks for sharing this example! I think the addition of another constructor might get confusing for internal use, as that public constructor would need to set a default value for the `internalRequiredParam`. This use case might be slightly different as it looks like we have an internal constructor for dependency injection purposes.